platform/mds/mds/src/main/java/org/motechproject/mds/annotations/Discriminated.java

```diff
@@ -0,0 +1,13 @@
+package org.motechproject.mds.annotations;
+
+import javax.jdo.annotations.IdentityType;
+import javax.jdo.annotations.PersistenceCapable;
+import java.lang.annotation.*;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@PersistenceCapable(identityType = IdentityType.DATASTORE, detachable = "true")
+public @interface Discriminated {
```

Comment:
Incidentally, @Discriminated may be a case where naming matters.  I feel that "discriminated" may not be the best name on the grounds of false association with the JDO feature by the name "discriminator", coincidentally used for implementation.  Here are some alternatives I have considered:

@Alternative
@Union
@Choice